### PR TITLE
Feature/fix fetch my contacts page page size

### PR DESCRIPTION
**Pull request by cedric25**
- fetchMyContacts() no more takes `page` and `pageSize` parameters: data come from two different sources at the moment so pagination will have to wait.

- Make two calls to Market API in parallel

- Have big pageSize parameter when calling Market API: Avoid too many round-trips as we need everything anyway.

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -15,7 +15,8 @@
   ],
   "rules": {
     "sonarjs/cognitive-complexity": "warn",
-    "@typescript-eslint/no-explicit-any": "warn"
+    "@typescript-eslint/no-explicit-any": "warn",
+    "@typescript-eslint/no-use-before-define": ["error", { "functions": false }]
```
Comment (cedric25) [inline .eslintrc.json:19]: Allow to define private functions at end of file (better for file readability)

File: src/web3mail/IExecWeb3mail.ts [hunk 2]
```diff
@@ -14,7 +13,7 @@ import { GraphQLClient } from 'graphql-request';
 import { DATAPROTECTOR_SUBGRAPH_ENDPOINT } from '../config/config.js';
 
 export class IExecWeb3mail {
-  fetchMyContacts: (args?: FetchContactsParams) => Promise<Contact[]>;
+  fetchMyContacts: () => Promise<Contact[]>;
```
Comment (cedric25) [inline src/web3mail/IExecWeb3mail.ts:16]: Cleaning `page` and `pageSize` parameters for now.

File: src/web3mail/fetchMyContacts.ts [hunk 3]
```diff
@@ -6,44 +6,27 @@ import { WorkflowError } from '../utils/errors.js';
...
+        iexec,
+        userAddress,
+        appAddress: WEB3_MAIL_DAPP_ADDRESS,
+      }),
+      fetchAllOrdersByApp({
+        iexec,
+        userAddress,
+        appAddress: WHITELIST_SMART_CONTRACT_ADDRESS,
+      }),
+    ]);
```
Comment (cedric25) [inline src/web3mail/fetchMyContacts.ts:29]: Extract a new function: `fetchAllOrdersByApp()` that does for each address:
 - Fetch the first page
 - Iterate to fetch all pages with `autoPaginateRequest()`

File: tests/e2e/fetchMyContacts.test.ts [hunk 4]
```diff
@@ -11,6 +11,11 @@ import { IExecWeb3mail, getWeb3Provider } from '../../dist/index';
 import { ValidationError, WorkflowError } from '../../dist/utils/errors';
 import { EnhancedWallet, IExec } from 'iexec';
 
+/**
+ * Run this test file:
+ * NODE_OPTIONS=--experimental-vm-modules npx jest tests/e2e/fetchMyContacts.test.ts --silent=false
+ */
```
Comment (cedric25) [inline tests/e2e/fetchMyContacts.test.ts:17]: Can clean if necessary. I usually like to have that at the top of my test files.

File: tests/e2e/fetchMyContacts.test.ts [hunk 5]
```diff
@@ -104,9 +109,7 @@ describe('web3mail.fetchMyContacts()', () => {
     const expectedErrorMessage =
       'Failed to fetch my contacts: wrong address is not a valid ethereum address';
 
-    await expect(web3mail.fetchMyContacts()).rejects.toThrow(
-      WorkflowError
-    );
+    await expect(web3mail.fetchMyContacts()).rejects.toThrow(WorkflowError);
```
Comment (cedric25) [inline tests/e2e/fetchMyContacts.test.ts:112]: Prettier...

File: tests/e2e/fetchMyContacts.test.ts [hunk 6]
```diff
@@ -117,9 +120,7 @@ describe('web3mail.fetchMyContacts()', () => {
       .spyOn(web3mail, 'fetchMyContacts')
       .mockRejectedValue(new ValidationError('Missing parameter'));
 
-    await expect(web3mail.fetchMyContacts()).rejects.toThrow(
-      ValidationError
-    );
+    await expect(web3mail.fetchMyContacts()).rejects.toThrow(ValidationError);
```
Comment (cedric25) [inline tests/e2e/fetchMyContacts.test.ts:123]: Prettier...

File: tests/unit/fetchMyContacts.test.ts [hunk 7]
```diff
@@ -6,9 +6,15 @@ import { Wallet } from 'ethers';
...
   DATAPROTECTOR_SUBGRAPH_ENDPOINT,
   WEB3_MAIL_DAPP_ADDRESS,
+  WHITELIST_SMART_CONTRACT_ADDRESS,
 } from '../../src/config/config';
 import { fetchMyContacts } from '../../dist/web3mail/fetchMyContacts';
 
+/**
+ * Run this test file:
+ * NODE_OPTIONS=--experimental-vm-modules npx jest tests/unit/fetchMyContacts.test.ts --silent=false
+ */
```
Comment (Le-Caignec) [inline tests/unit/fetchMyContacts.test.ts:16]: I'm not sure if it's very necessary to set this commentary as it's explicit in the package-lock.json. Nevertheless, if we set that we should set it in all the test files.

**Review by Le-Caignec — APPROVED** [review]

**Review by Le-Caignec — APPROVED** [review]

**Pull request merged**